### PR TITLE
qt: use upstream package folder architecture

### LINKED ISSUE #23660: [package] Qt/5.x: plugins dir for Android workaround
### Description

Currently there's [upstream bug](https://bugreports.qt.io/browse/QTBUG-113819) that prevents proper packaging of Android apps due to having the `plugins` dir in a non-standard location (Conan puts them in `bin/archdatadir/plugins`). As a local fix after building Qt I do:

```
cd package/<hash>
ln -s bin/archdatadir/plugins
```

I'd like to add the same workaround to the recipe, but not sure how to do it properly. Don't see how to create symlinks in https://docs.conan.io/2/reference/tools/files.html

### Package and Environment Details

* Package Name/Version: **qt/5.15.13**
* Operating System+version: **macOS 14.4.1 / Android**
* Compiler+version: **NDK r25c**
* Conan version: **conan 1.62.0**
* Python version: **Python 3.12.3**

### Conan profile

irrelevant

### Steps to reproduce

1. Build Qt 5 for android
2. build a Qt app for android
3. observe that Qt plugins aren't copied to the build directory, hence app fails to start

### Logs

irrelevant

### PR DESCRIPTION
**Pull request by kambala-decapitator**
Specify library name and version:  **qt/***

copy of #23944 by Eric's permission

fixes #23660

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by RubenRBS** [comment]
Thanks for rescuing this (and to Eric for allowing to use the commits), we'll check this tomorrow, try to move this forward asap :)

**Review by RubenRBS — COMMENTED** [review]
Got some local compilation issues for the recipe, would appreciate your insight!

File: recipes/qt/6.x.x/conanfile.py
Comment (kambala-decapitator) [reply recipes/qt/6.x.x/conanfile.py:869]: wanted to test on my side but Qt 6.7.1 build failed, will try with a different version later. Testing with v1 and
```
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=15
os=Macos
os_build=Macos
[options]
[build_requires]
[env]
[conf]
tools.cmake.cmaketoolchain:generator=Ninja
```
(Xcode 15.4)

Comment (RubenRBS) [reply recipes/qt/6.x.x/conanfile.py:869]: Hi! If using Conan 2.4, you might want to use this fix https://github.com/conan-io/conan-center-index/pull/24236 for it to compile (Or wait until it gets merged in a few hours/tomorrow and update this PR with the changes)

Comment (kambala-decapitator) [reply recipes/qt/6.x.x/conanfile.py:869]: in my case build error was from Qt, and that PR is about Windows :)
```
/Applications/Xcode-15.4.0.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/include/c++/v1/__memory/unique_ptr.h:66:19: error: invalid application of 'sizeof' to an incomplete type 'QQmlJS::Dom::BindingValue'
...
/Users/kambala/.conan/data/qt/6.6.3/_/_/source/src/qtdeclarative/src/qmldom/qqmldomelements_p.h:513:7: note: forward declaration of 'QQmlJS::Dom::BindingValue'
class BindingValue;
```

just built with `-o 'qt/*:essential_modules=False'` and all went fine, didn't face that error. Could you post exact conan command you issued and the exact assert message (i.e. which executable wasn't found)?

Also pipeline was green...

Comment (jcar87) [reply recipes/qt/6.x.x/conanfile.py:869]: > Also pipeline was green...

At the moment this is unfortunately misleading, because the recipe is currently set to build on CI with different options than a user would get, see: https://github.com/conan-io/conan-center-index/pull/23911

This is wrong and needs to be fixed - however because of the large amount of versions we are maintaining, this would result in very long CI builds.

Will try to fix this locally, the expectation is that with `-o 'qt/*:essential_modules=True'` the changes in this PR (which are otherwise great) should not result in failures

Comment (kambala-decapitator) [reply recipes/qt/6.x.x/conanfile.py:869]: @RubenRBS still unclear how you hit that assert and what was the exact message

Comment (jcar87) [reply recipes/qt/6.x.x/conanfile.py:869]: I get this too, with a simple
```
conan create . --version=6.7.1
```

keeping in mind that this causes `essential_modules=True` (by default)

Comment (jcar87) [reply recipes/qt/6.x.x/conanfile.py:869]: Why is this being changed from a warning to an assert in this PR?
On master it doesn't find this tool either `qmltestrunner `, but it is a warning.

Comment (jcar87) [reply recipes/qt/6.x.x/conanfile.py:871]: `qmltestrunner` is not a documented qt app as far as I can find, and it seems like its only built when certain conditions are met (including, potentially, the building of tests) - so I've added a commit to remove it from the list of checked executables, until we can work out if this is needed and how to fix this

as a side note, and I appreciate the changes on this PR existed previously, it is very difficult to review pull requests when there are changes that have nothing to do with the PR description :(

**Review by jcar87 — APPROVED** [review]

File: recipes/qt/6.x.x/conanfile.py
Comment (kambala-decapitator) [reply recipes/qt/6.x.x/conanfile.py:871]: Tried building with v2 and faced that error as well.

Also input from Eric:
> it is meant to avoid regressions, because this PR changed the folder in which executables are packaged
If this assert is triggered for executable qmltestrunner ,then someone should go take a look in the package folder to find where is is stored, and then add the path in this list https://github.com/conan-io/conan-center-index/pull/24193/files#diff-c548829da9d38ae15a8cb4447140bc50ac3f38250d1f348decd9569dc3298b24R862
> qmltestrunner may only be built when `-o qt/*:qtshadertools=True`

Comment (jcar87) [reply recipes/qt/6.x.x/conanfile.py:871]: `qmltestrunner` is not packaged - either before or after the changes in this PR, because it is not being built.  

The problem exists outside of the changes in this PR and it unrelated  - so let's address the package structure first.

**Review by RubenRBS — APPROVED** [review]

**Comment by SvenRoederer** [comment]
This change also fixes our problem with not finding the QtQuick-modules on application-startup. These modules get deployed in `qml/` subfolder, but are expected to be on top-level.

~~~
QQmlApplicationEngine failed to load component
qrc:/main.qml:7:1: module "QtDataVisualization" is not installed
qrc:/main.qml:6:1: module "Qt.labs.platform" is not installed
qrc:/main.qml:5:1: module "QtCharts" is not installed
qrc:/main.qml:4:1: module "QtQuick.Layouts" is not installed
qrc:/main.qml:3:1: module "QtQuick.Controls" is not installed
qrc:/main.qml:2:1: module "QtQuick.Window" is not installed
qrc:/main.qml:1:1: module "QtQuick" is not installed
qrc:/main.qml:7:1: module "QtDataVisualization" is not installed
qrc:/main.qml:6:1: module "Qt.labs.platform" is not installed
qrc:/main.qml:5:1: module "QtCharts" is not installed
qrc:/main.qml:4:1: module "QtQuick.Layouts" is not installed
qrc:/main.qml:3:1: module "QtQuick.Controls" is not installed
qrc:/main.qml:2:1: module "QtQuick.Window" is not installed
qrc:/main.qml:1:1: module "QtQuick" is not installed
qrc:/main.qml:7:1: module "QtDataVisualization" is not installed
qrc:/main.qml:6:1: module "Qt.labs.platform" is not installed
qrc:/main.qml:5:1: module "QtCharts" is not installed
qrc:/main.qml:4:1: module "QtQuick.Layouts" is not installed
qrc:/main.qml:3:1: module "QtQuick.Controls" is not installed
qrc:/main.qml:2:1: module "QtQuick.Window" is not installed
qrc:/main.qml:1:1: module "QtQuick" is not installed
qrc:/main.qml:7:1: module "QtDataVisualization" is not installed
qrc:/main.qml:6:1: module "Qt.labs.platform" is not installed
qrc:/main.qml:5:1: module "QtCharts" is not installed
qrc:/main.qml:4:1: module "QtQuick.Layouts" is not installed
qrc:/main.qml:3:1: module "QtQuick.Controls" is not installed
qrc:/main.qml:2:1: module "QtQuick.Window" is not installed
qrc:/main.qml:1:1: module "QtQuick" is not installed
qrc:/main.qml:7:1: module "QtDataVisualization" is not installed
qrc:/main.qml:6:1: module "Qt.labs.platform" is not installed
qrc:/main.qml:5:1: module "QtCharts" is not installed
qrc:/main.qml:4:1: module "QtQuick.Layouts" is not installed
qrc:/main.qml:3:1: module "QtQuick.Controls" is not installed
qrc:/main.qml:2:1: module "QtQuick.Window" is not installed
qrc:/main.qml:1:1: module "QtQuick" is not installed
qrc:/main.qml:7:1: module "QtDataVisualization" is not installed
qrc:/main.qml:6:1: module "Qt.labs.platform" is not installed
qrc:/main.qml:5:1: module "QtCharts" is not installed
qrc:/main.qml:4:1: module "QtQuick.Layouts" is not installed
qrc:/main.qml:3:1: module "QtQuick.Controls" is not installed
qrc:/main.qml:2:1: module "QtQuick.Window" is not installed
qrc:/main.qml:1:1: module "QtQuick" is not installed
qrc:/main.qml:7:1: module "QtDataVisualization" is not installed
qrc:/main.qml:6:1: module "Qt.labs.platform" is not installed
qrc:/main.qml:5:1: module "QtCharts" is not installed
qrc:/main.qml:4:1: module "QtQuick.Layouts" is not installed
qrc:/main.qml:3:1: module "QtQuick.Controls" is not installed
qrc:/main.qml:2:1: module "QtQuick.Window" is not installed
qrc:/main.qml:1:1: module "QtQuick" is not installed
~~~

With this change the modules are found below `qml/`.

**Pull request merged**
